python/tvm/relay/op/contrib/dnnl.py

```diff
@@ -76,12 +78,79 @@ def make_pattern(with_bias=True):
         conv_out = is_op("add")(conv, bias)
     else:
         conv_out = conv
-    return is_op("nn.relu")(conv_out)
+    if with_relu:
+        return is_op("nn.relu")(conv_out)
+    return conv_out
+
+
+def make_dense_pattern(with_bias=True, with_relu=True):
+    data = wildcard()
+    weight = wildcard()
+    bias = wildcard()
+    dense = is_op("nn.dense")(data, weight)
+    if with_bias:
+        dense_out = is_op("add")(dense, bias)
+    else:
+        dense_out = dense
+    if with_relu:
+        dense_out = is_op("nn.relu")(dense_out)
+    return dense_out
 
 
 @register_pattern_table("dnnl")
 def pattern_table():
-    conv2d_bias_relu_pat = ("dnnl.conv2d_bias_relu", make_pattern(with_bias=True))
-    conv2d_relu_pat = ("dnnl.conv2d_relu", make_pattern(with_bias=False))
-    dnnl_patterns = [conv2d_bias_relu_pat, conv2d_relu_pat]
+    conv2d_bias_relu_pat = (
+        "dnnl.conv2d_bias_relu",
+        make_conv_pattern(with_bias=True, with_relu=True),
+    )
+    conv2d_bias_pat = ("dnnl.conv2d_bias", make_conv_pattern(with_bias=True, with_relu=False))
+    conv2d_relu_pat = ("dnnl.conv2d_relu", make_conv_pattern(with_bias=False, with_relu=True))
+    dense_bias_pat = ("dnnl.dense_bias", make_dense_pattern(with_bias=True, with_relu=False))
+    dnnl_patterns = [
+        conv2d_bias_relu_pat,
+        conv2d_bias_pat,
+        conv2d_relu_pat,
+        dense_bias_pat,
+    ]
     return dnnl_patterns
+
+
+def partition_for_dnnl(
+    mod,
+    params=None,
+):
+    """Partition the graph greedily offloading supported operators to DNNL.
+
+    Parameters
+    ----------
+    mod : Module
+        The module to run passes on.
+    params : Optional[Dict[str, NDArray]]
+        Constant input parameters.
+    Returns
+    -------
+    mod : Module
+        Annotated and partitioned module.
+    """
+
+    if params:
+        mod["main"] = bind_params_by_name(mod["main"], params)
+    seq = tvm.transform.Sequential(
+        [
+            transform.CanonicalizeOps(),
+            transform.InferType(),
+            transform.SimplifyInference(),
+            transform.FoldConstant(),
+            transform.FoldScaleAxis(),
+            # fold consecutive add ops to simplify pattern `conv2d-bias_add-bn-relu`
+            transform.SimplifyExpr(),
+            transform.FoldConstant(),
+            transform.MergeComposite(pattern_table()),
+            transform.AnnotateTarget("dnnl"),
+            transform.MergeCompilerRegions(),
+            transform.PartitionGraph(),
+        ]
+    )
+    with tvm.transform.PassContext(opt_level=3):
```

Comment:
what's the difference result when the `opt_level` equals to values like 0/1/2? do you think user need this option when creating partition for dnnl backend?

Reply:
if `opt_level=1 / 2`, some passes like alter_op_layout will be disable, then it cannot pass the test cases, `opt_level=3` is the default setting.